data_structures/DynamicStack/DynamicStack.java

```
@@ -0,0 +1,144 @@
+        dStack.push(25);
+        stack.push(25);
+        dStack.pop();
+        stack.pop();
+        dStack.push(15);
+        stack.push(15);
+
+        try {
+            verifyEachElementInStacks(dStack, stack);
+        } catch(Exception err){
+            System.out.println("Basic test cases error thrown : "+err);
+        }
+    }   
+
+    
+    private void runRobustTestCases(){
+        DynamicStack dStack = new DynamicStack();
+        Stack<Integer> stack = new Stack<>();
+
+        int n = (int) Math.pow(10, 7);
+        for (int idx = 0; idx < n; idx++) {
+            dStack.push(idx);
+            stack.push(idx);
+            if (idx > 1 && (idx + 1) < n){
+                dStack.pop();
+                stack.pop();
+            }
+        }
+
+        try {
+            verifyEachElementInStacks(dStack, stack);
+        } catch(Exception err){
+            System.out.println("Robust test cases error thrown : "+err);
+        }
+    }
+
+    private void runInterLeaveTestCases() {
+        DynamicStack dStack = new DynamicStack();
+        Stack<Integer> stack = new Stack<>();
+
+        String[] operations = new String[]{"push", "pop"};
+        int n = (int) Math.pow(10, 7);
+        
+        for (int idx = 0; idx < n; idx++) {
+            int randomIndex = (int)(Math.random() * operations.length);
+            int randomNumber = (int)(Math.random() * 999) + 1; 
+
+            if (operations[randomIndex].equals("push")) { 
+                dStack.push(randomNumber);
+                stack.push(randomNumber);
+            }
+            
+            if (operations[randomIndex].equals("pop")) {
+                if (dStack.size() > 0) { 
+                    dStack.pop();
+                    stack.pop();
+                }
+            }
+        }
+        
+        try {
+            verifyEachElementInStacks(dStack, stack);
+        } catch(Exception err){
+            System.out.println("Interleave test cases error thrown : "+err);
```

Done. Swallow exceptions like this. Just call verify method() and let the exceptions bubble up.

---

You can use unchecked exceptions. For ex. Throw new RuntimeException(msg)